src/components/KonnectorFolder.jsx

```diff
@@ -1,25 +1,87 @@
-import React from 'react'
+import React, { Component, PureComponent } from 'react'
+import classNames from 'classnames'
+
+import Icon from 'cozy-ui/react/Icon'
 import { translate } from 'cozy-ui/react/I18n'
 
 import styles from '../styles/konnectorFolder'
 
 import DescriptionContent from './DescriptionContent'
 
-class KonnectorFolder extends React.Component {
-  render({ t, driveUrl, trigger }) {
+const ERRORED = 'ERRORED'
+const FETCHING = 'FETCHING'
+const SUCCESS = 'SUCCESS'
+
+/**
+ * Cache
+ */
+let driveUrl
+
+/**
+ * Renders a link only if href prop is provided
+ * @extends PureComponent
+ */
+class SmartLink extends PureComponent {
+  render() {
+    const { className, href } = this.props
+    return href ? (
+      <a className={className} href={href}>
+        {this.props.children}
+      </a>
+    ) : (
+      <span className={className}>{this.props.children}</span>
+    )
+  }
+}
+
+export class KonnectorFolder extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      status: FETCHING
+    }
+  }
+
+  async componentDidMount() {
+    if (!driveUrl) {
+      try {
+        const { client } = this.context
+        const { data } = await client.collection('io.cozy.apps').all()
+        const driveApp = data.find(app => app.slug === 'drive')
+        driveUrl = driveApp && `${driveApp.links.related}#/files/`
+      } catch (error) {
+        this.setState({
+          status: ERRORED
+        })
+      }
+    }
+
+    this.setState({
+      status: SUCCESS
+    })
```

Comment:
❗️ status will always be SUCCESS here since there is no early return in the catch

IMHO fetching the link should extracted from either the component or the componentDidMount.

The module variable should also be avoided.

```jsx
const memoizeUnless = (fn, checkMemoize) => {
  let res
  return function () {
    if (!checkMemoize(res)) {
     res = await fn.apply(this, arguments)
   }
   return res
}

const fetchDriveLink = memoizeUnless(client => {
     const { data } = await client.collection('io.cozy.apps').all()
     const driveApp = data.find(app => app.slug === 'drive')
     driveUrl = driveApp && `${driveApp.links.related}#/files/`
}, url => url !== undefined)

ensureDriveLink() {
  try {
   const driveURL = fetchDriveURL(this.context.client)
    this.setState({
          status: SUCCESS
     })
  } catch (error) {
    this.setState({
       status: ERRORED
     })
  }
  
}

async componentDidMount() {
  this.ensureDriveLink()
}
```

This way we have methods in the component only caring about the state of the component, and the method driveURL is easily testable.